op-batcher/batcher/channel_manager.go

```diff
@@ -199,6 +199,11 @@ func (s *channelManager) TxData(l1Head eth.BlockID) (txData, error) {
 		return txData{}, io.EOF
 	}
 
+	// we have blocks, but we cannot add them to the channel right now
+	if s.pendingChannel != nil && s.pendingChannel.IsFull() {
+		return txData{}, io.EOF
+	}
```

Comment:
this is now required because we call `TxData` when the `pendingChannel` is closed, which will return an "already closed" error when calling `outputFrames` below